src/tpg_tcp.c

```diff
@@ -835,6 +843,15 @@ inline bool tcp_send_ctrl_pkt_with_sseq(tcp_control_block_t *tcb, uint32_t sseq,
 
     stats = STATS_LOCAL(tpg_tcp_statistics_t, tcb->tcb_l4.l4cb_interface);
 
+    if (flags & RTE_TCP_SYN_FLAG)
```

Comment:
I'm a bit worried about the performance implications of rechecking the flags for every TCP packet. It might be better to increment the stats when we actually build the header.

Reply:
Yes I agree. These checks should not be made for TCP control packets that are pure ACKs only. I've modified the code flow so pure ACKs call into a separate function when sending.